File: reader/src/main/java/cn/archko/pdf/entity/DecodeParam.java
package cn.archko.pdf.entity;

import android.text.TextUtils;
import android.widget.ImageView;

import com.artifex.mupdf.fitz.Document;

import cn.archko.pdf.listeners.DecodeCallback;

/**
 * @author: archko 2024/3/17 :19:41
 */
public final class DecodeParam {
    public String key;
    public int pageNum;
    public float zoom;
    public int screenWidth;
    public ImageView imageView;
    public boolean crop;
    public int xOrigin;
    public APage pageSize;
    public Document document;
    public int targetWidth;
    public DecodeCallback decodeCallback;

    public DecodeParam(String key, int pageNum, float zoom, int screenWidth, ImageView imageView) {
        this.key = key;
        this.pageNum = pageNum;
        this.zoom = zoom;
        this.screenWidth = screenWidth;
        this.imageView = imageView;
    }

    public DecodeParam(String key, ImageView imageView, boolean crop, int xOrigin,
                       APage pageSize, Document document, DecodeCallback callback) {
        this.key = key;
        pageNum = pageSize.index;
        if (TextUtils.isEmpty(key)) {
            this.key = String.format("%s,%s,%s,%s", imageView, crop, xOrigin, pageSize);
        }
        this.imageView = imageView;
        this.crop = crop;
        this.xOrigin = xOrigin;
        this.pageSize = pageSize;
        this.document = document;
        this.decodeCallback = callback;
    }

    @Override
    public String toString() {
        return "DecodeParam{" +
                "key='" + key + '\'' +
                ", pageNum=" + pageNum +
                ", zoom=" + zoom +
                ", screenWidth=" + screenWidth +
                ", imageView=" + imageView +
                ", crop=" + crop +
                ", xOrigin=" + xOrigin +
                ", pageSize=" + pageSize +
                ", document=" + document +
                ", targetWidth=" + targetWidth +
                ", decodeCallback=" + decodeCallback +
                '}';
    }
}

File: readercommon/src/main/java/cn/archko/pdf/entity/DecodeParam.kt
package cn.archko.pdf.entity

import android.text.TextUtils
import android.widget.ImageView
import cn.archko.pdf.common.MupdfDocument
import cn.archko.pdf.listeners.DecodeCallback

/**
 * @author: archko 2024/3/17 :19:41
 */
class DecodeParam {
    var key: String
    var pageNum: Int
    var zoom = 0f
    var screenWidth = 0
    var imageView: ImageView
    var crop = false
    var xOrigin = 0
    var pageSize: APage? = null
    var document: MupdfDocument? = null
    var width = 1080
    var height = 1080
    var decodeCallback: DecodeCallback? = null

    constructor(key: String, pageNum: Int, zoom: Float, screenWidth: Int, imageView: ImageView) {
        this.key = key
        this.pageNum = pageNum
        this.zoom = zoom
        this.screenWidth = screenWidth
        this.imageView = imageView
    }

    constructor(
        key: String, imageView: ImageView, crop: Boolean, xOrigin: Int,
        pageSize: APage, document: MupdfDocument?, callback: DecodeCallback?,
        width: Int, height: Int
    ) {
        this.key = key
        pageNum = pageSize.index
        if (TextUtils.isEmpty(key)) {
            this.key = String.format("%s,%s,%s,%s", imageView, crop, xOrigin, pageSize)
        }
        this.imageView = imageView
        this.crop = crop
        this.xOrigin = xOrigin
        this.pageSize = pageSize
        this.document = document
        decodeCallback = callback
        this.width = width
        this.height = height
    }

    override fun toString(): String {
        return "DecodeParam{" +
                "key='" + key + '\'' +
                ", pageNum=" + pageNum +
                ", zoom=" + zoom +
                ", screenWidth=" + screenWidth +
                ", crop=" + crop +
                ", xOrigin=" + xOrigin +
                ", width=" + width +
                ", height=" + height +
                ", pageSize=" + pageSize +
                ", decodeCallback=" + decodeCallback +
                ", imageView=" + imageView +
                ", document=" + document +
                '}'
    }
}